### PR TITLE
178 publish collection use case

### LINKED ISSUE #178: Create use case for publishing a Collection
We need this for the Collection Page, which has an action for publishing a Collection

The API exists for this in Dataverse, https://guides.dataverse.org/en/latest/api/native-api.html#publish-dataverse-api

### PR DESCRIPTION
**Pull request by ekraffmiller**
## What this PR does / why we need it:
Creates a use case for publishing a collection
## Which issue(s) this PR closes:

- Closes #178 

## Related Dataverse PRs:

- Depends on #

## Suggestions on how to test this:
Inspect code and run tests.

### CODE REVIEW

**Review by g-saracca — DISMISSED** [review]
Looks good to me! Approving!
If you can also @GPortas make a quick code review while doing the QA will be great, thanks!

File: src/collections/domain/useCases/PublishCollection.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export class PublishCollection implements UseCase<void> {
+  private collectionsRepository: ICollectionsRepository
+
+  constructor(collectionsRepository: ICollectionsRepository) {
+    this.collectionsRepository = collectionsRepository
+  }
+
+  /**
+   * Publishes a collection, given its identifier and the type of version update type.
```
Comment (GPortas) [inline src/collections/domain/useCases/PublishCollection.ts]: I think the last part `"and the type of version update type."` should be removed!

**Review by GPortas — COMMENTED** [review]
Looks good! only one minor change related to a method doc. @ekraffmiller

File: src/collections/domain/useCases/PublishCollection.ts (hunk 1 above)
Comment (ekraffmiller) [reply src/collections/domain/useCases/PublishCollection.ts]: @GPortas thanks for finding that, done!

**Comment by g-saracca** [comment]
@ekraffmiller please merge develop into this branch before merging so we don't lose latest changes in file upload use case
cc: @GPortas

**Review by g-saracca — APPROVED** [review]

**Pull request merged**